framework/quota/src/org/apache/cloudstack/quota/vo/ServiceOfferingVO.java

```diff
@@ -310,14 +269,12 @@ public String getDeploymentPlanner() {
     }
 
     public String getDetail(String name) {
-        assert (details != null) : "Did you forget to load the details?";
-
-        return details != null ? details.get(name) : null;
+        if (details == null) throw new IllegalStateException("Cannot set value as details is null");
+        return details.get(name);
     }
 
     public void setDetail(String name, String value) {
-        assert (details != null) : "Did you forget to load the details?";
-
+        if (details == null) throw new IllegalStateException("Cannot set value as details is null");
```

Comment:
Do you really like this kind of solution? A POJO  with some kind of verification does not seem like the best thing to have.

Why don’t we initialize the as an empty map? This way it would be always ready for use, and we would not need that verification. We could change line 81 to Map<String, String> details = new HashMap<String, String> ();